server/modules/protocol/MySQL/mysql_common.cc

```diff
@@ -1768,3 +1767,126 @@ void mxs_mysql_execute_kill_user(MXS_SESSION* issuer, const char* user, kill_typ
 
     mxs_mysql_send_ok(issuer->client_dcb, info.targets.size(), 0, NULL);
 }
+
+/**
+ *  Decode ok packet and get session track info
+ *
+ *  @param buffer               Buffer contain ok packet
+ *  @param server_capabilities  Server capabilities
+ *
+ */
+void mxs_mysql_get_session_track_info(GWBUF *buff, uint32_t server_capabilities)
+{
+    char *trx_info, *var_name, *var_value;
+
+    if (GWBUF_LENGTH(buff) < 9  || !mxs_mysql_is_ok_packet(buff))
+    {
+        return;
+    }
+
+    if (!GWBUF_IS_CONTIGUOUS(buff))
```

Comment:
This could be replaced by copying the buffer data into a stack allocated buffer:
```
size_t len = gwbuf_length(buff);
char local_buf[len];
gwbuf_copy_data(buff, 0, len, local_buf);
```
This way you have the data in a contiguous area of memory but you skip the extra call to `malloc` that is done when the buffer is made contiguous.